client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoDriverUri.java

```diff
@@ -31,11 +32,24 @@ public final class TrinoDriverUri
 {
     private static final String JDBC_URL_PREFIX = "jdbc:";
     private static final String JDBC_URL_START = JDBC_URL_PREFIX + "trino:";
+    private static final String LEGACY_PREPARED_STATEMENTS = "LegacyPreparedStatements";
+    private Optional<Boolean> legacyPreparedStatements = Optional.empty();
+
+    private static Properties trinoProperties(Properties driverProperties)
+    {
+        Properties newProperties = (Properties) driverProperties.clone();
+        newProperties.remove(LEGACY_PREPARED_STATEMENTS);
+        return newProperties;
+    }
```

Comment:
This remove may be inlined in the super call?

Reply:
Is there a method in Java which performs a remove in the result without modifying the original map?